app/routes/v2/clusters.js

```diff
@@ -122,13 +122,25 @@ var runAddClusterWebhook = async(req, orgId, clusterId, clusterName)=>{
   }
 };
 
-function pushToS3Sync(key, searchableDataHash, dataStr, data_location, logger) {
+function pushToS3Sync(org, key, searchableDataHash, dataStr, data_location, logger) {
   //if its a new or changed resource, write the data out to an S3 object
   const result = {};
-  const bucket = conf.storage.getResourceBucket(data_location);
+  // const bucket = conf.storage.getResourceBucket(data_location);
+  const locationConf = conf.storage.s3ConnectionMap.get(data_location || conf.storage.defaultLocation) || {};
+  const bucketPrefix = locationConf.orgBucketPrefix || 'razee-org-';
+  const bucket = `${bucketPrefix}${org._id.toLowerCase()}`;
+  // var bucketKey = `${data_location}-resources`;
+  const bucketConfObj = {
+    type: 'active',
+    region: data_location,
+    kind: 'resources',
+  };
   const hash = crypto.createHash('sha256');
   const keyHash = hash.update(JSON.stringify(key)).digest('hex');
-  const handler = storageFactory(logger).newResourceHandler(`${keyHash}/${searchableDataHash}`, bucket, data_location);
+  const path = `${org._id}/${keyHash}/${searchableDataHash}`;
+  console.log(22222, data_location, locationConf, org);
+  const factory = storageFactory({ logger });
+  const handler = factory.newResourceHandler({ path, bucketConfObj, data_location, org });
```

Comment:
Same comment as earlier: avoid duplicate data_location in both the conf obj and the new handler function signature.